app/src/main/java/com/simplecity/amp_library/playback/MusicService.java

```diff
@@ -849,14 +849,17 @@ public void toggleRepeat() {
         switch (getRepeatMode()) {
             case QueueManager.RepeatMode.OFF:
                 setRepeatMode(QueueManager.RepeatMode.ALL);
+                newUiEvent(queueManager.getCurrentSong(), UiEventType.REPEAT_ALL_SONGS);
```

Comment:
Is there a reason we wouldn't capture UI events in the fragments themselves? Seems risky to be capture UI events in the Service in case something goes wrong in the Fragment/Service communication.

Reply:
Ah - after looking at the code, it looks like the UI doesn't actually hold the repeat state, so you need to get this from the service?

Reply:
In the Repeat button's `onClick` function, only one `toggleRepeat` function is called and the classification of the repeat modes happens inside of the service. However, I agree that this could be quite dicey if the communication between was cut off for some reason. 
Alternatively, we could create a function that uses the `getRepeatMode` function from the  `MusicService.java` and returns the current RepeatMode and record the events accordingly on the `PlayerFragment`. Let me know what you think?

Reply:
@barbeau Correct - Sorry, I had the above comment written before Github updated me with your second comment.

Reply:
Ok, thanks. I think it's fine as-is then.